### PR TITLE
feat: add sessionName in the attach to session dropdown

### PR DESCRIPTION
**Pull request by huzaifaiftikhar**
Once #1693 is merged users will be able to view the list of running sessions in the attach to session window. It is difficult to filter the sessions via the session-id, therefore, adding the `sessionName` (if present) to the list of `importantCaps`.

**Screenshot from the dev build**
<kbd>sessionName in attach to session list</kbd>

### CODE REVIEW

**Comment by KazuCocoa** [comment]
What's the `sessionName`?
It seems the value is not a regular Appium stuff.

**Comment by KazuCocoa** [comment]
btw, please sign the CLA

**Comment by huzaifaiftikhar** [comment]
> What's the `sessionName`?
> It seems the value is not a regular Appium stuff.

Yes, it's not an appium capability, this capability is specific to BrowserStack by which we allow the user's to give a meaningful name to their sessions.
Would it possible to accept this PR as a feature request or do you think otherwise?

**Comment by KazuCocoa** [comment]
Could it be possible to build your own custom AD for your platform with this change?

It would be better to define a new module or method to allow a vendor to customize the naming for them.
I don't check around the code well, but something like:

```js
function formatCaps (vendor, caps) {
    if (vendor === 'browserstack') {
        return formatCapsBrowserStack(caps)
    }
    // ...
}

function formatCapsBrowserStack (caps) {
    // do your own logic
}
```

Or extract the place as a different module.

**Comment by huzaifaiftikhar** [comment]
@KazuCocoa can you re-review? I've created the method `formatCapsBrowserstack` and other cloud providers can change the formatting as per their choice using a similar approach.

File: app/renderer/components/Session/AttachToSession.js [hunk 1]
```diff
@@ -15,14 +16,26 @@ function formatCaps (caps) {
...
+  }
+  return importantCaps.join(', ').trim();
+}
+
 export default class AttachToSession extends Component {
 
-  getSessionInfo (session) {
-    return `${session.id} — ${formatCaps(session.capabilities)}`;
+  getSessionInfo (session, serverType) {
+    if (serverType === ServerTypes.browserstack) {
```
Comment (KazuCocoa) [inline app/renderer/components/Session/AttachToSession.js:30]: Could you change here to `switch`?

File: app/renderer/components/Session/AttachToSession.js [hunk 2]
```diff
@@ -15,14 +16,26 @@ function formatCaps (caps) {
   return importantCaps.join(', ').trim();
 }
 
+function formatCapsBrowserstack (caps) {
```
Comment (KazuCocoa) [inline app/renderer/components/Session/AttachToSession.js:19]: 👍

File: app/renderer/components/Session/AttachToSession.js (hunk 1 above)
Comment (huzaifaiftikhar) [reply app/renderer/components/Session/AttachToSession.js:30]: Done

**Review by KazuCocoa — APPROVED** [review]

**Pull request merged**